src/frontend/components/UI/SearchBar/index.tsx

```diff
@@ -1,63 +1,32 @@
 import { Search } from '@mui/icons-material'
-import React, {
-  useCallback,
-  useContext,
-  useEffect,
-  useMemo,
-  useRef
-} from 'react'
-import { useTranslation } from 'react-i18next'
-import { useNavigate } from 'react-router-dom'
-import ContextProvider from 'frontend/state/ContextProvider'
+import React, { Fragment, useCallback, useEffect, useRef } from 'react'
 import './index.scss'
 import { faXmark } from '@fortawesome/free-solid-svg-icons'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { GameInfo } from '../../../../common/types'
 
-function fixFilter(text: string) {
-  const regex = new RegExp(/([?\\|*|+|(|)|[|]|])+/, 'g')
-  return text.replaceAll(regex, '')
+interface Props {
+  suggestionsListItems?: JSX.Element[]
+  onInputChanged: (text: string) => void
+  value: string
+  placeholder: string
 }
 
-const RUNNER_TO_STORE = {
-  legendary: 'Epic',
-  gog: 'GOG',
-  nile: 'Amazon'
-}
-
-export default React.memo(function SearchBar() {
-  const { handleSearch, filterText, epic, gog, sideloadedLibrary, amazon } =
-    useContext(ContextProvider)
-  const { t } = useTranslation()
-  const navigate = useNavigate()
-
+export default function SearchBar({
```

Comment:
I made this component more generic, customizable with props to extend it for both LibrarySearchBar and WinetricksSearchBar, most of the code I removed here is now in LibrarySearchBar